editoast/editoast_models/src/model/error.rs

```diff
@@ -0,0 +1,58 @@
+use std::sync::LazyLock;
+
+use diesel::result::DatabaseErrorKind;
+use regex::Regex;
+
+#[derive(Debug, thiserror::Error)]
+pub enum Error {
```

Comment:
Do you think we would need a `CreateError` and a `DeleteError`? From what I understand, they semantically can produce different set of error variants... but `diesel` seems to have made the choice to only have one `enum` for them all (even if different functions may only return a subset of it).

Reply:
https://github.com/OpenRailAssociation/osrd/pull/10196#discussion_r1905522292

Reply:
I'm not sure that I understand how is this related to https://github.com/OpenRailAssociation/osrd/pull/10196#discussion_r1905522292.